src/main/java/io/appform/dropwizard/discovery/bundle/id/IdGenerator.java

```diff
@@ -164,8 +196,12 @@ public static Optional<Id> generateWithConstraints(String prefix, String domain)
      * @param skipGlobal Skip global constrains and use only passed ones
      * @return Id if it could be generated
      */
-    public static Optional<Id> generateWithConstraints(String prefix, String domain, boolean skipGlobal) {
-        return generateWithConstraints(prefix, DOMAIN_SPECIFIC_CONSTRAINTS.getOrDefault(domain, Collections.emptyList()), skipGlobal);
+    public static Optional<Id> generateWithConstraints(String prefix, @NonNull String domain, boolean skipGlobal) {
+        if (!REGISTERED_DOMAINS.containsKey(domain)) {
```

Comment:
@sarb1208 This would be a breaking change from current implementation POV

Reply:
removed